app/models/stripe_payment_intent.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+# == Schema Information
+#
+# Table name: payment_intentions
+#
+#  id           :bigint(8)        not null, primary key
+#  code         :string           not null
+#  reference    :string           not null
+#  category     :string           default("stripe"), not null
+#  payable_type :string
+#  payable_id   :bigint(8)
+#  metadata     :jsonb            not null
+#
+
+class StripePaymentIntent < ApplicationRecord
```

Comment:
My understanding of a plugin would be that code doesn't need to be mixed into the main codebase. Would this be an exception to that or should this file be placed inside the gem?

Reply:
Ah this is a file generated by the gem actually. This file is there in the generator of the gem.
we can run rails generator code to generate this code and migration with it.
rails generate ecko:plugins:stripe:payment_intention will generate this file.